qa/rpc-tests/wallet_z_sendmany.py

```diff
@@ -230,5 +284,70 @@ def run_test(self):
         self.check_balance(0, 0, n0ua0, {'sapling': 8})
         assert_equal(Decimal(self.nodes[2].z_getbalance(myzaddr)), zbalance)
 
+        #
+        # Test that z_sendmany avoids UA linkability unless we allow it.
+        #
+
+        # Generate a new account with two new addresses.
+        n1account = self.nodes[1].z_getnewaccount()['account']
+        n1ua0 = self.nodes[1].z_getaddressforaccount(n1account)['unifiedaddress']
+        n1ua1 = self.nodes[1].z_getaddressforaccount(n1account)['unifiedaddress']
+
+        # Send funds to the transparent receivers of both addresses.
+        for ua in [n1ua0, n1ua1]:
+            taddr = self.nodes[1].z_listunifiedreceivers(ua)['transparent']
+            self.nodes[0].sendtoaddress(taddr, 2)
+
+        self.sync_all()
+        self.nodes[2].generate(1)
+        self.sync_all()
+
+        # The account should see all funds.
+        assert_equal(
+            self.nodes[1].z_getbalanceforaccount(n1account)['pools'],
+            {'transparent': {'valueZat': 4 * COIN}},
+        )
+
+        # The addresses should see only the transparent funds sent to them.
+        assert_equal(self.nodes[1].z_getbalance(n1ua0), 2)
+        assert_equal(self.nodes[1].z_getbalance(n1ua1), 2)
+
+        # If we try to send 3 ZEC from n1ua0, it will fail with too-few funds.
+        recipients = [{"address":n0ua0, "amount":3}]
+        linked_addrs_msg = 'Insufficient funds: have 2.00, need 3.00 (This transaction may require selecting transparent coins that were sent to multiple Unified Addresses, which is not enabled by default because it would create a public link between the transparent receivers of these addresses. THIS MAY AFFECT YOUR PRIVACY. Resubmit with the `privacyPolicy` parameter set to `AllowLinkingAccountAddresses` or weaker if you wish to allow this transaction to proceed anyway.)'
```

Comment:
Does the returned message always include a policy that is weak enough, or just one based on the first error that it detects?

If the latter, I guess it needn't block this PR, but we should have an issue to make it more precise.

Reply:
The latter. Making it more precise would require a large-scale refactor of the backend to support a commit-apply approach where we first figure out all the effects of the transaction, and then apply them. Note that currently half of the checks need to occur during the async part, due to how `z_sendmany` runs note selection in that phase.

Reply:
Filed #5703. I think we should make a web page (for the time being it can be just a copy of the RPC doc, we can expand it later) and refer to that in the error message. That does not need to block 4.7.0-rc1.